pkg/ebpf/lockcontention.go

```diff
@@ -0,0 +1,492 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-present Datadog, Inc.
+
+//go:build linux_bpf
+
+package ebpf
+
+import (
+	"errors"
+	"fmt"
+	"math"
+	"os"
+	"slices"
+	"sync"
+	"syscall"
+	"unsafe"
+
+	"github.com/DataDog/datadog-agent/pkg/ebpf/bytecode"
+	"github.com/DataDog/datadog-agent/pkg/util/kernel"
+	"github.com/DataDog/datadog-agent/pkg/util/log"
+	manager "github.com/DataDog/ebpf-manager"
+
+	"github.com/cilium/ebpf"
+	"github.com/cilium/ebpf/link"
+	"github.com/prometheus/client_golang/prometheus"
+)
+
+const (
+	// TrackAllEBPFResources decides if all system ebpf resources should be collected
+	// or just system-probe resources
+	TrackAllEBPFResources = true
```

Comment:
Yeah I think it is useful to have this true by default. This would allow us to track contention issues on programs outside the scope of system-probe as well, for example cilium programs.

Reply:
It is worth noting that we do not track external ebpf resources in the ebpf check due to the cardinality issue